src/MovieCard/MovieCard.js

```diff
@@ -4,42 +4,48 @@ import { Link, Redirect } from "react-router-dom";
 import { postRating, deleteRatingApi } from "../apiCalls/apiCalls";
 
 class MovieCard extends Component {
-	constructor({userID, title, averageRating, poster, id, userRatings, getUsersRatings}, props) {
+	constructor(props) {
 		super()
 		this.state = {
 			foundRating: null,
 			rating: null,
 			clicked: false,
 			error: null,
 		}
-		this.userID = userID
-		this.title = title
-		this.averageRating = averageRating;
-		this.poster = poster;
-		this.id = id;
-		this.userRatings = userRatings;
-		this.getUsersRatings = getUsersRatings;
 	}
 
 	componentDidMount = () => {
-		if (this.userRatings) {
-		  if (this.userRatings.find((rating) => rating.movie_id === this.id)) {
-        	let foundRating = this.userRatings.find((rating) => rating.movie_id === this.id);
-        	this.setState({ foundRating: foundRating.rating });
-      		}
-		}
+		this.checkForUserRating();
 	}
 
-	submitRating = (event) => {
+	submitRating = async (event) => {
 		event.preventDefault();
-		postRating(Number(this.state.rating), this.id, this.userID)
-			.then(() => {this.getUsersRatings(this.userID)})
-			.catch((error) => this.setState({ error }));
+		try {
+			await postRating(Number(this.state.rating), this.props.id, this.props.userID)
+			await this.props.getUsersRatings(this.props.userID); 
```

Comment:
similar situation -- we need to await the fetch call coming from App so our data is in sync w/ page loads